### PR TITLE
Emanuele Gorga - Nov 2018: Chitter Challenge

### PR DESCRIPTION
**Pull request by emanuelegorga**
I had lot of fun doing this weekend challenge. I feel more confident in the all process and I tried to do all the functionality of the app so I left behind the style. I did all the user story, also the one in which you can tag someone it looks like working but I am not sure wether or not it sends an email. 

- I'd like you to focus on my methodical approach and how I can get better at it.

### CODE REVIEW

File: spec/features/peep_spec.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+require_relative 'web_helper'
+
+feature 'User can post a peep' do
+  scenario 'Shows one peep on the page' do
+    post_peep
+    expect(page).to have_content 'Writing a peep'
+  end
+end
```
Comment (scass91) [inline spec/features/peep_spec.rb]: It's good to see tests, but it'd be even better to see the tests pushed before they pass so you can prove your workflow is methodical. Remember: Failing test, push, passing test, push, refactor, push, repeat.

File: spec/features/testing_infrastructure_spec.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
-    visit('/')
-    expect(page).to have_content 'Testing infrastructure working!'
-  end
-end
+# feature 'Testing infrastructure' do
+#   scenario 'Can run app and check page content' do
+#     visit('/')
+#     expect(page).to have_content 'Testing infrastructure working!'
+#   end
+# end
```
Comment (scass91) [inline spec/features/testing_infrastructure_spec.rb]: Would this test still work if uncommented?

File: README.md [hunk 3]
```diff
@@ -53,72 +45,45 @@ So that I can stay constantly tapped in to the shouty box of Chitter
...
+3. You can see your peeps in your profile page or have a look at all peeps created by other users in the index.
 
-You can see your test coverage when you run your tests. If you want this in a graphical form, uncomment the `HTMLFormatter` line and see what happens!
+Technical Approach and technologies I used:
+-----
+* sinatra - To create the web app using Ruby;
+* postgresql - Database;
+* database_cleaner - Cleans up the database for each test ran;
+* data_mapper - Ruby ORM;
+* capybara - Test frameworks used for web developent.
```
Comment (scass91) [inline README.md]: It's good to have the tech used here as you can refer back to it later - Consider adding any diagramming which may have happened to show your methodical workflow.

File: lib/peep.rb [hunk 4]
```diff
@@ -1,4 +1,5 @@
 require 'data_mapper'
+require_relative 'user'
```
Comment (scass91) [inline lib/peep.rb]: I think you only need require 'data_mapper' here

File: lib/user.rb [hunk 5]
```diff
@@ -2,12 +2,28 @@
 require 'dm-validations'
 require 'dm-postgres-adapter'
```
Comment (scass91) [inline lib/user.rb]: Similar to peep.rb, remove references to gems and put them in your gemfile (if they're not there already). Also retain require 'data_mapper'

File: spec/features/web_helper.rb [hunk 6]
```diff
@@ -2,9 +2,10 @@ def sign_up
   visit('/')
   click_on 'Sign Up'
   expect(page.current_path).to eq('/signup')
-  fill_in :username, with: 'myusername'
   fill_in :email, with: 'myemail@email.com'
   fill_in :password, with: 'mypassword'
+  fill_in :name, with: 'myname'
+  fill_in :username, with: 'myusername'
   click_on 'Submit'
 end
```
Comment (scass91) [inline spec/features/web_helper.rb]: Add as much to this as necessary, it makes tests look a lot cleaner!

File: spec/features/testing_infrastructure_spec.rb (hunk 2 above)
Comment (emanuelegorga) [reply spec/features/testing_infrastructure_spec.rb]: Obviusly not. This was just the first feature test to test if the infrastructure was actually working.

File: Gemfile [hunk 7]
```diff
@@ -1,11 +1,21 @@
 source 'https://rubygems.org'
 
-ruby '2.5.0'
+ruby '2.6.0'
```
Comment (becc-mu) [inline Gemfile]: Your Ruby version is 2.5.0, but your Gemfile specified 2.6.0

Comment (emanuelegorga) [reply Gemfile]: screenshot 2019-01-26 at 08 44 05
As in the screenshot, I double checked and can confirm that I updated  the ruby version on my laptop before doing the Chitter Challenge, so at the time of the challenge it was already 2.6.0.

**Pull request closed without merging**